core/metacling/src/TCling.cxx

```diff
@@ -4046,7 +4047,7 @@ void TCling::SetClassInfo(TClass* cl, Bool_t reload)
    // that is currently in the caller (like SetUnloaded) that disable AutoLoading and AutoParsing and
    // code is in the callee (disabling template instantiation) and end up with a more explicit class:
    //      TClingClassInfoReadOnly.
-   TClingClassInfo* info = new TClingClassInfo(GetInterpreterImpl(), name.c_str(), instantiateTemplate);
+   TClingClassInfo* info = (preChecked && decl && !static_cast<const clang::Decl *>(decl)->isInvalidDecl()) ? new TClingClassInfo(GetInterpreterImpl(), decl) : new TClingClassInfo(GetInterpreterImpl(), name.c_str(), instantiateTemplate);
```

Comment:
I don't think with need the `preChecked` boolean at all.  Either the passed `decl` is set (it was pre-checked and found) or it is not set (it was not pre-checked or was not-precheked and not found but still somehow we want to create `TClingClassInfo`); in all those cases we would just want to executed the old/pre-existing code as it was.